### DISCUSSION TITLE
where is  stored  a saved password?

### DISCUSSION THREAD

**Discussion by faustf** [dbeaver/dbeaver#20815] [Q&A] [answered]
### Description

Hi i have linux mint 64  with dbever 23.1.3.202307231901 last version flat pack , i not remember the  password of my mysql  but , i saved inside of my dbever , and  i can enter inside  , but i want  save  the passowrd  stored in dbever , where is ??  

thanks so much

### DBeaver Version

23.1.3.202307231901

### Operating System

linux mint 64bit

### Database and driver

mysql  Ver 8.0.33-0ubuntu0.22.04.4 for Linux on x86_64 ((Ubuntu))

### Steps to reproduce

connect database

### Additional context

by dbevere

**Comment by serge-rider** [comment, marked as answer]
Short answer: it is not possible to get saved password from DBeaver UI in the Community edition.

Long answer:

#### Community edition
Password and other credentials are saved in file `credentials-config.json`. It is encrypted file. It is possible to unencrypt this file because no strong cryptography is used in the Community version. 

#### PRO editions
- Depending on the worspace configuration passwords are saved in the same `credentials-config.json` file or in special encrypted secure storage (not in DBeaver workspace), there is no easy way to unencrypt this storage if you don't know the master password.
- You can set DBeaver project's password (this will encrypt database cridentials with strong cryptography algorithms) - than you can use button "Show password". You have to enter project password to see database password.

See https://dbeaver.com/docs/wiki/Project-security/

If you don't have PRO version of DBeaver then you can:
- Install trial version of any PRO product:
- Temporarily set project password
- See connection password
- Remove project password
